### ISSUE TITLE
[Efficiency] User encounters unnecessary repetition or redundancy.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#75978]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** govcio-vfep
**Product:** GI Bill Comparison Tool
**Feature:** Specialized Mission Filter

## Findings details
**VA.gov Experience Standard - issue:** User encounters unnecessary repetition or redundancy.
**VA.gov Experience Standard - category:** Efficiency
**Launch-blocking:** Yes
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

In the Community Focus section, screen reader users hear the "Go to community focus details" link text whenever any of the checkboxes receives focus. This doesn't match what the checkbox actually does and could be confusing and distracting to users --- especially since the order in which the text is announced will put "Go to community focus details" *before* each label, so users will have to wait through that text string to hear the actual checkbox label. I think this is likely to prevent at least some screen reader users from successfully filtering on the community focus criteria.

<details><summary>Link, screenshot or steps to recreate</summary>

Steps to reproduce:

1. Navigate to the [GI Bill Comparison Tool on staging](https://staging.va.gov/education/gi-bill-comparison-tool/).
2. Activate a screen reader (recommend VoiceOver on Mac or NVDA on Windows for testing).
3. Press [tab] until the checkbox labeled "Historically Black Colleges and Universities" receives focus. Note that the screen reader announces "Go to community focus details" as part of its message, eg. on NVDA: `Community focus, Go to community focus details, Historically Black Colleges and Universities, checkbox unchecked`.
4. Press [tab] one time to move to the next checkbox. Note that the screen reader continues to announce the link text, eg. `Community focus, Go to community focus details, Men's colleges and universities`.

</details>

## Recommended action
This is happening because each checkbox references the element that contains the heading and the link via `aria-labelledby`:
```
   <span id="checkbox-group-84-legend" class="gibct-legend">
      <h3 class="school-types-label" aria-level="2">Community focus</h3>
      <button class="mobile-jump-link labels-margin">
         <a class="jump-link arrow-down-link" href="#learn-more-about-specialized-missions-accordion-button" tabindex="0">
            <p>
               <span>Go to community focus details</span>
            </p>
          </a>
      </button>
   </span>
```
and
```
   <div class="vads-l-col--4 my-filters-margin">
      <input id="checkbox-group-88-0" name="specialMissionHbcu" data-testid="special-mission-hbcu" type="checkbox" aria-labelledby="checkbox-group-88-legend specialmissionhbcu-0-label">
      <label class="gi-checkbox-label vads-u-margin-right--1p5" id="specialmissionhbcu-0-label" name="specialMissionHbcu-label" for="checkbox-group-88-0">Historically Black Colleges and Universities</label>
   </div>
```

As in the other issue logged about this `aria-labelledby` pattern, I think this could largely be solved via a more typical `fieldset`/`legend` pattern rather than using `aria-labelledby`. But at a minimum, move the "Go to community focus details" link out of the `#checkbox-group-84-legend` element so it's not announced at every checkbox.

## References

- **WCAG Success Criteria:** [WCAG 2.5.3 Label in Name - Level A](https://www.w3.org/TR/WCAG21/#label-in-name)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arianaadili** [comment]
Our team's test engineer confirmed the changes in the staging environment on 2/28. JAWS/NVDA navigates through each of the community focus elements without announcing the phrase 'Go to' in advance.

**Comment by arianaadili** [comment]
Enclosed is a screen recording created by our front-end dev team, showcasing the utilization of a screen reader and voice-over assistive technology to present every element on the page.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/b7373b4b-4916-48e0-9b42-de866a8596f9
